content/posts/a-guide-on-commit-messages.mdx

```diff
@@ -3,26 +3,35 @@ title: A guide on commit messages
 date: 2020-01-29
 ---
 
-In this article we will talk about writing good commit messages not good commits even if these two subject are closely linked. Here is a [good article](https://www.git-tower.com/blog/version-control-best-practices/) from Tobias Günther that summarize well best practices about version control. 
+The time you had an archive on your disk called "version 1" should be far away. I bet you use some tool for version control like GitHub, Gitlab, Bitbucket, ... So you must write commits everytime you want to add your code to a project. Did you ever ask yourself if your commits messages are good? Maybe you don't like writing them and wonder why you have to write it every time you want to push code. After all the change must be sufficient enough? 
+
+In this article, we will talk about writing good commit messages not good commits even if these two subjects are closely linked (here is a [good article](https://www.git-tower.com/blog/version-control-best-practices/) from Tobias Günther that summarize well best practices about version control).
 
 ## Why commit message are important
 
-- First way to communicate change to fellows developers
-  - The diff is the what the commit message is the why
-- Make a lot of git commands much more useful
-  - Review changes without other tools than git (git log -p)
-  - Catch up when coming back from holidays using git log
-  - Make rebase less random
-  - Don't know why a line of code is made like that use git blame (it is not made to insult your co-workers)
-- Speed up reviewing process
-  - Reduce time to get context of a change (this time is mutiplied by the number of reviewer)
-- Generate nice changelog (changes between two versions)
-- Make projects easier to maintain in long term
-  - With a good commit message it is easier to catch up why some change was made five years ago
+So, is the change is sufficient? Obviously, the answer is no. But why?
+
+Commit messages have a lot of usages. This is the first way to communicate with your fellow developers on a project. The change is the "What" the commit message is the "Why". It must give enough context to a developper that might ask himself why some code is written like that. When a commit message gives you enough context about the change you won't spend precious time asking yourself the reason a developer push this code. In some case like open source projects busy maintainers won't even read your code if you don't give them a minimum of information like a nicely written commit message.
```

Comment:
The change is the "What" the commit message is the "Why".
Imo this sentence is not necessary and can lead to some misunderstanding

Reply:
I changed it. Is it more clear now?

Reply:
It's a bit more clear but I'm still not found of this point of view as you know.

Reply:
I'm not sure I understand the sentence either

Reply:
To be clear, I mean this one: `The change is the "What" the commit message is the "Why"`

Reply:
I changed this sentence to something more clear imo

Reply:
Oh! My bad, I found the new version and it's way better 😉

Reply:
So I resolve this thread 👍